src/transpiling/jsx_precompile.rs

```diff
@@ -444,70 +563,98 @@ impl JsxPrecompile {
+
+          let expr = self.gen_template(index, strings, dynamic_exprs);
+          elems.push(Some(ExprOrSpread {
+            spread: None,
+            expr: Box::new(expr.clone()),
+          }));
+        } else {
+          // Here we parse children the normal way
+
+          for child in normalized_children.iter() {
+            match child {
+              // Case: <div>foo</div>
+              JSXElementChild::JSXText(jsx_text) => {
+                elems.push(Some(ExprOrSpread {
+                  spread: None,
+                  expr: Box::new(string_lit_expr(jsx_text.value.to_string())),
+                }));
               }
-            }
-            // Case: <div><span /></div>
-            JSXElementChild::JSXElement(jsx_el) => {
-              let expr = self.serialize_jsx(jsx_el);
-              elems.push(Some(ExprOrSpread {
-                spread: None,
-                expr: Box::new(expr.clone()),
-              }));
-            }
-            // Case: <div><></></div>
-            JSXElementChild::JSXFragment(jsx_frag) => {
-              if let Some(child_expr) =
-                self.serialize_jsx_children_to_expr(&jsx_frag.children)
-              {
-                match child_expr {
-                  Expr::Array(array_lit) => {
-                    for item in array_lit.elems.iter() {
-                      elems.push(item.clone());
-                    }
-                  }
-                  _ => {
+              // Case: <div>{2 + 2}</div>
+              JSXElementChild::JSXExprContainer(jsx_expr_container) => {
+                match &jsx_expr_container.expr {
+                  // Empty JSX expressions can be ignored as they have no content
+                  // Case: <div>{}</div>
+                  // Case: <div>{/* some comment */}</div>
+                  JSXExpr::JSXEmptyExpr(_) => continue,
+                  JSXExpr::Expr(expr) => {
                     elems.push(Some(ExprOrSpread {
                       spread: None,
-                      expr: Box::new(child_expr.clone()),
+                      expr: expr.clone(),
                     }));
                   }
-                };
+                }
+              }
+              // Case: <div><span /></div>
+              JSXElementChild::JSXElement(jsx_el) => {
+                let expr = self.serialize_jsx(jsx_el);
+                elems.push(Some(ExprOrSpread {
+                  spread: None,
+                  expr: Box::new(expr.clone()),
+                }));
               }
+              // Case: <div><></></div>
+              JSXElementChild::JSXFragment(jsx_frag) => {
+                if let Some(child_expr) =
+                  self.serialize_jsx_children_to_expr(&jsx_frag.children)
+                {
+                  match child_expr {
+                    Expr::Array(array_lit) => {
+                      for item in array_lit.elems.iter() {
+                        elems.push(item.clone());
+                      }
```

Comment:
Can this just be `elems.extend(array_lit.elems)`?

Reply:
Ohh nice! Didn't know about this :+1: